akd/agents/aspect_search/aspect_search.py

```diff
@@ -0,0 +1,198 @@
+                                                      survey_subjects,
+                                                      route_messages)
+from akd.agents.aspect_search.structures import (InterviewState,
+                                                 Perspectives,
+                                                 update_references, 
+                                                 update_search_results)
+from akd.tools.search import SearchTool, SearchResultItem, SearxNGSearchTool
+
+
+class AspectSearchInputSchema(InputSchema):
+    """Input schema for aspect search agent"""
+    topic: Union[str, List[str]] = Field(..., description="Topic to search for.")
+
+
+class AspectSearchOutputSchema(OutputSchema):
+    """Output schema for aspect search agent"""
+    search_results: Union[List[SearchResultItem], List[List[SearchResultItem]]] = Field(None, description="List of search results returned by the search engine after the interviews.")
+    references: Union[Dict[str, str], List[Dict[str, str]]] = Field(None, description="List of references.")
+    perspectives: Union[Perspectives, List[Perspectives]] = Field(None, description="List of perspectives used to explore the topic.")
+
+
+class AspectSearchConfig(BaseAgentConfig):
+    """Configuration for Aspect Search Agent"""
+    retry_attempts: Optional[int] = Field(default=3, 
+                                description="Number of retry attempts.")
+    num_editors: Optional[int] = Field(default=3, 
+                             description="Number of editors to participate in the interview")
+    max_turns: Optional[int] = Field(default=3, 
+                           description="Maximum number of turns each interview runs for.")
+    top_n_wiki_results: Optional[int] = Field(default=3, 
+                                    description="Number of wiki results to return to generate perspectives")
+    max_wiki_ctx_len: int = Field(default=1500,
+                                 description="Maximum length of wiki content context for perspective generation.")
+    search_tool: Optional[object] = Field(default=SearxNGSearchTool(), 
+                                    description="Search tool to use.")
+    category: Optional[str] = Field(default=None, 
+                          description="Category for the search tool.")
+    max_ctx_len: int = Field(default=15000,
+                                 description="Maximum length of the search result context during interviews.")
+
+
+class AspectSearchAgent(BaseAgent):
+    input_schema = AspectSearchInputSchema
+    output_schema = AspectSearchOutputSchema
+    config_schema = AspectSearchConfig
+
+    def _post_init(
+        self,
+    ) -> None:
+        super()._post_init()
+
+        self.llm = ChatOpenAI(
+             model=self.config.model_name, temperature=self.config.temperature, api_key=self.config.api_key
+        )
+
+        self.wikipedia_retriever = WikipediaRetriever(load_all_available_meta=True, 
```

Comment:
TODO for future:
Change retriever to inject from constructor for better flexiblity.